lib/utils/fetch.ts

```diff
@@ -1,38 +1,52 @@
 import { Auth } from 'aws-amplify';
 
-// https://eckertalex.dev/blog/typescript-fetch-wrapper
-async function http<T>(path: string, config?: RequestInit): Promise<T> {
+export type ResponseWithStatus<T> = { data: T; isSuccess: true } | { isSuccess: false; errorMessage: string };
+
+async function http<T>(path: string, config?: RequestInit): Promise<ResponseWithStatus<T>> {
     const authSession = await Auth.currentSession();
     const jwt = authSession.getIdToken()?.getJwtToken();
     const request = new Request(path, {
         headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${jwt}` },
         ...config,
     });
-    const response = await fetch(request);
-
-    if (!response.ok) {
```

Comment:
It's probably not a good idea _not_ to clearly communicate this so that it's accessible from the components. The `data` field is unreliable since .e.g `!""==true`